src/main/java/com/exactpro/th2/codec/grpc/ProtoUtils.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.th2.codec.grpc;
+
+import java.util.List;
+import java.util.Map;
+
+import com.exactpro.th2.common.grpc.ListValue;
+import com.exactpro.th2.common.grpc.Message;
+import com.exactpro.th2.common.grpc.Value;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import org.jetbrains.annotations.NotNull;
+
+public class ProtoUtils {
+	public static Message toMessage(DynamicMessage dynamicMessage) {
+		Message.Builder messageBuilder = Message.newBuilder();
+		messageBuilder.getMetadataBuilder().setMessageType(dynamicMessage.getDescriptorForType().getName());
+		
+		for (Map.Entry<Descriptors.FieldDescriptor, Object> fieldEntry : dynamicMessage.getAllFields().entrySet()) {
+			Descriptors.FieldDescriptor descriptor = fieldEntry.getKey();
+			Object objectValue = fieldEntry.getValue();
+			Value value = convertToValue(objectValue);
+			
+			messageBuilder.putFields(descriptor.getName(), value);
+		}
+		return messageBuilder.build();
+	}
+
+    private static Value convertToValue(Object fieldValue) {
+        Value.Builder valueBuilder = Value.newBuilder();
+        if (fieldValue instanceof com.google.protobuf.Message) {
+            Message nestedMessage = convertComplex((com.google.protobuf.Message) fieldValue);
+            valueBuilder.setMessageValue(nestedMessage);
+        } else if (fieldValue instanceof List<?>) {
+            ListValue listValue = convertToListValue(fieldValue);
+            valueBuilder.setListValue(listValue);
+        } else {
```

Comment:
Please add certain check for types of simple fields and throw an exception otherwise